### ISSUE TITLE
feature: navigating between flows

### ISSUE THREAD

**Issue by waynehamadi** [windmill-labs/windmill#3817]
use command + click to go to a flow, and command + arrow left to go back. Particularly important in the vscode extension because I can't even see a subflow when I click on it.
more info here
https://www.loom.com/share/b256b791105649a3aa55cfc7b2cc122a

**Comment by keeganpoppen** [comment]
oOo that is a very neat idea

**Comment by keeganpoppen** [comment]
there's other stuff in that vein that i think could be kind of fire... like, you should be rewarded for decomposing your flows into calling other flows, but it kinda turns into a disaster area. at least the details view and stuff when you run the thing works admirably well for how complex the problem is.

**Comment by waynehamadi** [comment]
exactly, this is common across the whole low-code industry: the more you modularize, the worse your experience is.

This is a very hard problem to solve though.

**Comment by rubenfiszel** [comment]
This will be addressed this month ("expand" Button to expand a flow step by the flow itself)

**Comment by keeganpoppen** [comment]
well that's why i wanted to give props to the details view for when a flow
runs, because it has no right to work as well as it does-- it's great! and
it really is a very, very difficult problem. as far as i see it, there are
a couple high-level directions to go in to alleviate it:
- make it easier to transition more flows / more of the windmill infra to
the rest of your own code & infra, so that the windmill parts become more
of a pointer to the code that used to live in windmill but you're now
pretty happy with, reliability-wise and runs with the rest of your system.
this turns windmill more in the direction of a prototyping tool where flows
"graduate" to the rest of your system over time. very cool, but i would
understand if it doesn't serve what windmill views as their target
market/customer, as it would be positioning as a "prototyping tool for
power users" (this is, however, exactly how i use the product haha). this
probably involves finding a way to unify their workers w/ things like the
FaaS paradigm more directly (which is the direction i've been taking
lately-- i have a lot of flows that just have side effects spawning NATS
NEX services & workers (basically Cloudflare Workers) so that i can keep my
beloved windmill flows where they are happy-- with mostly static structure,
not too many levels of nesting, and using the stuff Windmill is great as to
the fullest extent.
- allow some sort of insane infinitely-zooming canvas + somehow still
making the UI obvious enough that you are transitioning between different
~domains wrt. versioning and stuff. would be cool, but almost certainly a
complete quagmire involving multiple problems that have never been solved
before (for a reason). there are probably "cheap" ways to get part of the
way there, like rendering the full flows so that you can expand/contract
them, click on them, etc., but it really is a very dangerous game to let
you freely edit a flow within a flow within a flow carte blanche, unless
you are fixing those inner flows at certain hashes (because if you are
doing this it is more likely to be ok to branch off from that flow's
version history and make a bespoke version of the inner flow for this
purpose).
- basically exactly what they are doing currently, which is to allow you to
make behemoths like you and i do wayne while it obviously being a strain on
your and their system to do so, but doing a great job with showing you what
actually happened on your run in the details & graphs screens. which they
absolutely do. let's just say i can understand why this is the current
efficient frontier for them at the moment. but i can tell from their
releases that they definitely care about #1 (e.g. with code bundling,
bringing in your own react components, etc.).

at the end of the day, there aren't that many examples where a flow works
well both standalone and as like a subflow of a subflow-- usually you start
with a few small ones, nail them down, then build big ones that become
flows / features in their own right. i'm sure there are ways to smooth the
difficulty curve, but there's probably a reason why no one that i've seen
has "solved" it yet (cf. Chesterton's Fence).

btw, stoked to see the change Ruben is talking about!
